### PR TITLE
hw-jv-work-with-file

### PR DESCRIPTION
**Pull request by ihor-sydorenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,8 +1,66 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private int buy = 0;
+    private int supply = 0;
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:12]: lets make this variables local

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,8 +1,66 @@
...
 
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(toFileName, createReport(readFromFile(fromFileName)));
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:15]:
```suggestion
       String fileData = readFromFile(fromFileName);
       String report = createReport(fileData);
       writeToFile(toFileName, report);
```

 `writeToFile(toFileName, createReport(readFromFile(fromFileName)));` it is really hard to read

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,8 +1,66 @@
...
+        for (String resultStrings : strings) {
+            stringBuilder.append(resultStrings).append(COMMA);
+        }
+        return stringBuilder.toString();
+    }
+
+    private String[] createReport(String dataFromFile) {
+        int result = 0;
+        String[] report = null;
+        String[] split = dataFromFile.split(COMMA);
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:36]:
```suggestion
        String[] splittedFileData = dataFromFile.split(COMMA);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,8 +1,66 @@
...
+                + "buy," + buy + System.lineSeparator() + "result," + result};
+        return report;
+    }
 
+    private void writeToFile(String fileName, String[] report) {
+        File file = new File(fileName);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't create file", e);
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:56]:
```suggestion
            throw new RuntimeException("Can't create file " + fileName, e);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,8 +1,66 @@
...
+        }
+        for (String operation : report) {
+            try {
+                Files.write(file.toPath(), operation.getBytes(), StandardOpenOption.APPEND);
+            } catch (IOException e) {
+                throw new RuntimeException("Can't write data to file", e);
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:62]:
```suggestion
                throw new RuntimeException("Can't write data to file " + fileName, e);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,8 +1,66 @@
...
+                supply = supply + Integer.parseInt(split[i + 1]);
+            }
+            result = supply - buy;
+        }
+        report = new String[]{"supply," + supply + System.lineSeparator()
+                + "buy," + buy + System.lineSeparator() + "result," + result};
+        return report;
+    }
 
+    private void writeToFile(String fileName, String[] report) {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:51]:
```suggestion
    private void writeToFile(String fileName, String report) {
```

lets create ready report in the form of a String in `createReport()` method

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**